trl/trainer/ppo_trainer.py

```diff
@@ -488,12 +492,17 @@ def generate(
 
             if length_sampler is not None:
                 generation_kwargs["max_new_tokens"] = length_sampler()
-            response = self.accelerator.unwrap_model(self.model).generate(
-                input_ids=query_tensor.unsqueeze(dim=0), **generation_kwargs
-            )
+
+            with unwrap_model_for_generation(self.model, self.accelerator) as unwrapped_model:
+                response = unwrapped_model.generate(input_ids=query_tensor.unsqueeze(dim=0), **generation_kwargs)
+
             if generate_ref_response:
-                with self.optional_peft_ctx():
```

Comment:
I've moved the logic for disabling the adapter to `unwrap_model_for_generation()`

Also, why did we do the adapter disabling for the reference model but not the active model?